lib/puppetserver/ca/local_certificate_authority.rb

```diff
@@ -36,11 +36,30 @@ class LocalCertificateAuthority
         ["authorityKeyIdentifier", "keyid:always", false]
       ].freeze
 
+      attr_reader :cert, :key, :crl
+
       def initialize(digest, settings)
         @digest = digest
         @host = Host.new(digest)
         @settings = settings
         @errors = []
+
+        if ssl_assets_exist?
+          @cert, @key, @crl = load_ssl_assets
+        end
+      end
+
+      def ssl_assets_exist?
+        File.exist?(@settings[:cacert]) &&
+          File.exist?(@settings[:cakey]) &&
+          File.exist?(@settings[:cacrl])
+      end
+
+      def load_ssl_assets(settings=@settings)
+        cert = OpenSSL::X509::Certificate.new(File.read(settings[:cacert]))
+        key = OpenSSL::PKey::RSA.new(File.read(settings[:cakey]))
+        crl = OpenSSL::X509::CRL.new(File.read(settings[:cacrl]))
+        return cert, key, crl
```

Comment:
It might be good to continue to use the X509Loader here, since it has a lot of nice validation and error messages. I still think it's valuable to tell users when their CA is not set up correctly, especially when all the files exist, because that probably indicates that something serious went wrong.